From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas R <atoomic@cpan.org>
Date: Fri, 15 Feb 2019 17:19:20 -0700
Subject: [PATCH 28/30] Avoid a warning when prompting install_help intro

Case CPANEL-25679:

view GH #124

install_help does not have one intro and this would
raise a warning in some weird cases.
---
 SOURCES/perl/cpan/CPAN/lib/App/Cpan.pm       | 2 +-
 SOURCES/perl/cpan/CPAN/lib/CPAN/FirstTime.pm | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/SOURCES/perl/cpan/CPAN/lib/App/Cpan.pm b/SOURCES/perl/cpan/CPAN/lib/App/Cpan.pm
index 80c3efec..2a1ad491 100644
--- a/SOURCES/perl/cpan/CPAN/lib/App/Cpan.pm
+++ b/SOURCES/perl/cpan/CPAN/lib/App/Cpan.pm
@@ -676,7 +676,7 @@ sub _hook_into_CPANpm_report
 
 	*CPAN::Shell::myprint = sub {
 		my($self,$what) = @_;
-		$scalar .= $what;
+		$scalar .= $what if defined $what;
 		$self->print_ornamented($what,
 			$CPAN::Config->{colorize_print}||'bold blue on_white',
 			);
diff --git a/SOURCES/perl/cpan/CPAN/lib/CPAN/FirstTime.pm b/SOURCES/perl/cpan/CPAN/lib/CPAN/FirstTime.pm
index 177b0fb6..273688c2 100644
--- a/SOURCES/perl/cpan/CPAN/lib/CPAN/FirstTime.pm
+++ b/SOURCES/perl/cpan/CPAN/lib/CPAN/FirstTime.pm
@@ -1697,7 +1697,8 @@ sub my_prompt_loop {
     my $ans;
 
     if (!$auto_config && (!$m || $item =~ /$m/)) {
-        $CPAN::Frontend->myprint($prompts{$item . "_intro"});
+        my $intro = $prompts{$item . "_intro"};
+        $CPAN::Frontend->myprint($intro) if defined $intro;
         $CPAN::Frontend->myprint(" <$item>\n");
         do { $ans = prompt($prompts{$item}, $default);
         } until $ans =~ /$ok/;
